tests/integration/vlan_test.py

```diff
@@ -100,8 +100,8 @@ def test_rollback_for_vlans(eth1_up):
         libnmstate.apply(desired_state)
 
     time.sleep(5)  # Give some time for NetworkManager to rollback
-    current_state_after_apply = libnmstate.show()
-    assert current_state == current_state_after_apply
+
+    assertlib.assert_state(current_state)
```

Comment:
Now we do not know if the dns or routes have been affected by the rollback.
Another option is to normalize the states are remove the metric from routes before the comparison. On that function, we can add the bug as a docstring.

Reply:
Since DNS and route configured are stored in interface config, there should be no difference
on NM rollback. We are not testing whether NM API works as it advertised but nmstate API.

Meanwhile I said in other PR, adding route support to `assert_state` require complex code to
handle merge/absent/sort staff which a waste of effort as we already have those tests in `route_test.py`.

We already have [bug 1738269][1] to track this issue. And this bug require zero changes in
nmstate, I think there is no need to track it in our CI test.


[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1738269

Reply:
The integration tests are checking the whole path, including NM and even the kernel.
Unit tests check only Nmstate, integration tests checks everything.

There are bugs/limitations in NM that we implement workarounds in nmstate (e.g. IPv6 enable/disable).
Eventually, we do test NM as well, there is no way we can run out of it.

What I do agree with in this regard is that the tests should be focused. If we just want to check the interfaces state, we should not fail on routes and DNS. On the other hand, I do want to show that in case of a rollback, some thing do not work.
Therefore, I am fine with adjusting the test as you propose but I would like to have a test that shows the problem as well and mark it with (soft) `xfail` referencing to the BZ.

Reply:
That make sense. I will create the xfail test case.